onejs/index.ts

```diff
@@ -61,4 +61,34 @@ export function eventfulSignal<T, K extends keyof T>(obj: T, propertyName: K, ev
         removeEventFunc.call(obj, onValueChangedCallback)
     })
     return sig
+}
+
+/**
+ * Run a callback when a C# event fires. The callback will be cleaned up on
+ * unmount and when the OneJS engine reloads.
+ * @param obj The object that the C# event is attached to.
+ * @param eventName The variable name of the C# event.
+ * @param callback The callback to run when the event fires.
+ * @param dependencies: The dependencies to pass to useEffect. Previous versions
+ * of the callback will be cleaned up any dependency changes.
+ */
+export function useEvent<TEventName extends string, TCallback>(
+  obj: Record<
+    `add_${TEventName}` | `remove_${TEventName}`,
+    (callback: TCallback) => void
+  >,
+  eventName: TEventName,
+  callback: TCallback,
+  dependencies: any[] = []
+) {
+  const remove = obj[`remove_${eventName}`].bind(obj);
+  function removeHandler() {
+    remove(callback);
+  }
+
+  useEffect(() => {
+    obj[`add_${eventName}`](callback);
+    onEngineReload(removeHandler);
```

Comment:
One issue here is that when a dependency causes `useEffect` to re-evaluate its callback, the remove handler will be added again. There is currently nothing in the OneJS API that will allow us to unregister the reload handler, e.g. an `unsetOnEngineReload`. But that would be nice!